jax/_src/numpy/lax_numpy.py

```diff
@@ -6285,11 +6539,12 @@ def wrapped(*args, **kwargs):
 # functions, which can themselves handle instances from any of these classes.
 
 _scalar_types = (int, float, complex, np.generic)
+_accepted_binop_types = (int, float, complex, np.generic, np.ndarray, ndarray)
 
 def _defer_to_unrecognized_arg(binary_op):
   # Ensure that other array types have the chance to override arithmetic.
   def deferring_binary_op(self, other):
-    if not isinstance(other, _scalar_types + _arraylike_types + (core.Tracer,)):
+    if not isinstance(other, _accepted_binop_types):
```

Comment:
This is a change of behavior, no? Because some tracers may not have an arraylike aval?

Reply:
Yes, this would now reject Tracers that are not arrays. But does anyone pass such a thing to one of these operators — wouldn't these all only work for arrays anyway?

In practice, there are no test failures in our internal tests from this cause.